# TODO: make as maps

# SDSO settings
playbackSpeed: 0
preload: 0

# TODO: do we need useSampleOutput here?
mode: 1
quiet: 0
nogui: 0
# TODO: change the path to local
calib: "camera_zed2_radtan.txt" # needs to be a file name
left_image_topic: /zed2/zed_node/left/image_rect_color
right_image_topic: /zed2/zed_node/right/image_rect_color

setting_desiredImmatureDensity: 1500
setting_desiredPointDensity: 2000
setting_minFrames: 5
setting_maxFrames: 7
setting_maxOptIterations: 6
setting_minOptIterations: 1
setting_kfGlobalWeight: 0.3
setting_maxShiftWeightT: 40
setting_maxShiftWeightR: 40
setting_maxShiftWeightRT: 20
setting_minGradHistAdd: 15 # 15: selects points on the edges very well
setting_logStuff: false

# RadiusOutlierRemoval settings
useFiltering: false
activeMinNeighborsInRadius: 4 # 8
activeRadiusSearch: 0.2 # 0.2
marginMinNeighborsInRadius: 4
marginRadiusSearch: 0.2

# For localization
publishMarginCloud: false
accumulateGlobalCloud: false
useReferenceCloud: true
referenceCloudFile: "sample_reference_cloud.pcd"

# --- NOT USED ---

minNumPointsToSend: 100
useRANSAC: false

# RANSAC parameters
distanceThreshold: 0.1 # 0.01
probability: 0.99  # NOT IN USE
maxIterations: 1000  # NOT IN USE

# StatisticalOutlierRemoval settings
meanK: 10
stddevMulThresh: 0.01

# Clustering parameters
clusterTolerance: 0.2 # 0.02
minClusterSize: 35
maxClusterSize: 5000
